modules/material/themes/material/default/selectidp-links.twig

```diff
@@ -0,0 +1,129 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <title>{{ '{selectidp-links:title}'|trans }}</title>
+
+  {% include 'header.twig' %}
+
+  <script>
+    function setSelectedIdp(id) {
+      const idpInput = document.createElement('input');
+
+      idpInput.type = 'hidden';
+      idpInput.name = '{{ returnIDParam|e }}';
+      idpInput.value = id;
+
+      document.querySelector('form').appendChild(idpInput);
+
+      ga('send', 'event', 'hub', 'choice', 'IdP', id);
+    }
+
+    function clickedAnyway(idpName) {
+      ga('send', 'event', 'hub', 'choice-disabled', 'IdP', idpName);
+    }
+  </script>
+</head>
+<body>
+<div class="mdl-layout mdl-layout--fixed-header fill-viewport">
+  <header class="mdl-layout__header">
+    <div class="mdl-layout__header-row">
+      <span class="mdl-layout-title scale-to-parent">
+        {% if spName ?? '' is empty %}
```

Comment:
SSP must have `strict_variables` set to true because I have seen errors for undefined variables. I haven't settled on the best place to have the null coalesce. I'll try to make sure the controller defines everything that's expected. It might be tough to ensure that, though, especially shared template files like the header and footer.

Reply:
Oh, in this particular case, it's a work in progress because `spName` isn't defined by the standard discovery service. So it definitely will be undefined at present.